### PR TITLE
Remove underscore usage

### PR DESCRIPTION
**Pull request by bernhardoj**
### Details

Removing underscore usage from the codebase.

### Fixed Issues

$ https://github.com/Expensify/App/issues/39121
PROPOSAL: https://github.com/Expensify/App/issues/39121#issuecomment-2058887964

### Tests

Same as QA Steps
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as QA Steps
### QA Steps

1. Open the money request page
2. Enter any amount and press Next
3. Verify the Recent and Contact lists are shown correctly
4. Press Split to add a user to the split participant
5. Verify the user is added
6. Press the checked icon to unchecked it
7. Verify the user is unchecked
8. Press Split on a workspace and a user
9. Verify the Add to split button is disabled and there is an error message
10. Search for an unexisting user
11. Verify the unexisting user is shown
12. Select the user
13. Verify the user is shown on the confirmation page
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/50919443/8c5b0a3f-5878-4b10-9b87-22c1e4dfc945

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by blazejkustra** [comment]
@bernhardoj Lint is failing 👀

**Comment by blazejkustra** [comment]
Couple things that I noticed:
- We should remove all `// eslint-disable-next-line rulesdir/prefer-underscore-method` comments from the codebase now.
- You didn't uninstall underscore from the repo
- Let's remove underscore from `CIGitLogicTest.sh` too
- If needed let's use lodash in JS files to unblock this PR (in proptypes files and all remaining)
- How a bout adjusting all comments in the codebase that refers to underscore? Example: `// underscore promises in sequence without for-loop`

File: src/SCREENS.ts [hunk 1]
```diff
@@ -265,7 +265,6 @@ const SCREENS = {
     },
 
     EDIT_REQUEST: {
-        ROOT: 'EditRequest_Root',
```
Comment (blazejkustra) [inline src/SCREENS.ts:268]: Why is this removed in this PR?

File: src/components/OptionsSelector/BaseOptionsSelector.js [hunk 2]
```diff
@@ -77,9 +78,9 @@ class BaseOptionsSelector extends Component {
         this.calculateAllVisibleOptionsCount = this.calculateAllVisibleOptionsCount.bind(this);
         this.handleFocusIn = this.handleFocusIn.bind(this);
         this.handleFocusOut = this.handleFocusOut.bind(this);
-        this.debouncedUpdateSearchValue = _.debounce(this.updateSearchValue, CONST.TIMING.SEARCH_OPTION_LIST_DEBOUNCE_TIME);
+        this.debouncedUpdateSearchValue = lodashDebounce(this.updateSearchValue, CONST.TIMING.SEARCH_OPTION_LIST_DEBOUNCE_TIME);
         this.relatedTarget = null;
-        this.accessibilityRoles = _.values(CONST.ROLE);
+        this.accessibilityRoles = Object.values(CONST.ROLE);
```
Comment (blazejkustra) [inline src/components/OptionsSelector/BaseOptionsSelector.js:83]: Let's disable the lint error with a comment

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Comment by hungvu193** [comment]
Since we will have 2,3 reviewers here, @bernhardoj can you also include my [suggestion here](https://github.com/Expensify/App/issues/39121#issuecomment-2060168273) in this PR? 
Besides @blazejkustra's [comment here](https://github.com/Expensify/App/pull/40346#issuecomment-2061440422), we need to do some additional tasks.
- [ ] Remove underscore in eslint and webpack config 
- [ ] Remove (Update) underscore usage from our [contributing guides](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md)

File: src/components/OptionsSelector/BaseOptionsSelector.js (hunk 2 above)
Comment (hungvu193) [reply src/components/OptionsSelector/BaseOptionsSelector.js:83]: I think it would be better if we removed the ESLint config.

Comment (bernhardoj) [reply src/components/OptionsSelector/BaseOptionsSelector.js:83]: Or should i just use lodash here?

File: src/SCREENS.ts (hunk 1 above)
Comment (bernhardoj) [reply src/SCREENS.ts:268]: EditRequestPage uses underscore but the page itself isn't being used anymore, so I remove it too.

File: src/components/OptionsSelector/BaseOptionsSelector.js (hunk 2 above)
Comment (bernhardoj) [reply src/components/OptionsSelector/BaseOptionsSelector.js]: Updated to use lodash

**Comment by bernhardoj** [comment]
> * We should remove all `// eslint-disable-next-line rulesdir/prefer-underscore-method` comments from the codebase now.
> * You didn't uninstall underscore from the repo
> * Let's remove underscore from `CIGitLogicTest.sh` too
> * If needed let's use lodash in JS files to unblock this PR (in proptypes files and all remaining)
> * How a bout adjusting all comments in the codebase that refers to underscore? Example: `// underscore promises in sequence without for-loop`

Done, but after removing the underscore from `CIGitLogicTest.sh`, the unit test fails. @blazejkustra Do you know how to fix it? Or is it expected?

image


> Remove underscore in eslint and webpack config

@hungvu193 Removed from eslint, but I don't see any underscore-related config in webpack.

> Remove (Update) underscore usage from our [contributing guides](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md)

@mountiny Do we need to update this?

File: .eslintrc.js [hunk 3]
```diff
@@ -238,7 +238,6 @@ module.exports = {
                 'jsdoc/no-types': 'error',
                 'rulesdir/no-default-props': 'error',
                 'import/no-extraneous-dependencies': 'off',
-                'rulesdir/prefer-underscore-method': 'off',
```
Comment (bernhardoj) [inline .eslintrc.js:241]: @blazejkustra We got lint errors when I removed this. Do we want to update `eslint-config-expensify` to remove this rule or keep turning this rule off?

Comment (hungvu193) [reply .eslintrc.js:241]: I feel like we should `eslint-config-expensify`.

Comment (bernhardoj) [reply .eslintrc.js:241]: @mountiny can you confirm?

_I will update this tomorrow_

Comment (bernhardoj) [reply .eslintrc.js]: I have turned this off and added back `npm install underscore` in `CIGitLogicTest` for now since we don't have any answer yet. I think you can start reviewing it. @hungvu193

Comment (bernhardoj) [reply .eslintrc.js]: Btw, I ask for the help in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1713523057689269)

File: src/SCREENS.ts (hunk 1 above)
Comment (mountiny) [reply src/SCREENS.ts:273]: Can you clarify how come its not used yet? what other pages are used to edit request?

Comment (bernhardoj) [reply src/SCREENS.ts:273]: I searched for the usage of the edit request route and component and found nothing.

It's previously used when we are editing money request fields, but we have combined the edit and create page into one like in this [PR](https://github.com/Expensify/App/pull/39723).

File: .eslintrc.js (hunk 3 above)
Comment (hungvu193) [reply .eslintrc.js]: Sure. I'll review in a while

File: contributingGuides/STYLE.md [hunk 4]
```diff
@@ -112,38 +112,9 @@ if (someCondition) {
...
-const modifiedArray = _.chain(someArray)
-    .filter(filterFunc)
-    .map(mapFunc)
-    .value();
-```
-
 ## Accessing Object Properties and Default Values
 
-Use `lodashGet()` to safely access object properties and `||` to short circuit null or undefined values that are not guaranteed to exist in a consistent way throughout the codebase. In the rare case that you want to consider a falsy value as usable and the `||` operator prevents this then be explicit about this in your code and check for the type using an underscore method e.g. `_.isBoolean(value)` or `_.isEqual(0)`.
+Use `lodashGet()` to safely access object properties and `||` to short circuit null or undefined values that are not guaranteed to exist in a consistent way throughout the codebase. In the rare case that you want to consider a falsy value as usable and the `||` operator prevents this then be explicit about this in your code and check for the type.
```
Comment (hungvu193) [inline contributingGuides/STYLE.md:117]: I'm not sure if this is still correct since we added `prefer-nullish-coalescing` rule. cc @mountiny

File: contributingGuides/STYLE.md [hunk 5]
```diff
@@ -112,38 +112,9 @@ if (someCondition) {
 }
 ```
 
-## Object / Array Methods
-
-We have standardized on using [underscore.js](https://underscorejs.org/) methods for objects and collections instead of the native [Array instance methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods). This is mostly to maintain consistency, but there are some type safety features and conveniences that underscore methods provide us e.g. the ability to iterate over an object and the lack of a `TypeError` thrown if a variable is `undefined`.
```
Comment (hungvu193) [inline contributingGuides/STYLE.md:117]: I think we should update this to `standardized on using lodash or native method` instead of deleting it

**Comment by hungvu193** [comment]
We have conflicts here @bernhardoj

File: contributingGuides/STYLE.md (hunk 4 above)
Comment (bernhardoj) [reply contributingGuides/STYLE.md:117]: This is a guideline for JS Style, I think we should remove it but there are few outdated guideline in this file, so not sure. (not related to underscore too)

File: contributingGuides/STYLE.md (hunk 5 above)
Comment (bernhardoj) [reply contributingGuides/STYLE.md:117]: We have `you-dont-need-lodash-underscore`, so pretty sure we favor to use native object/array method. For example, instead of `lodashKeys`, we prefer `Object.keys`.

**Comment by bernhardoj** [comment]
Conflicts solved.

> 1. Waiting for https://github.com/Expensify/App/pull/35822 before we can remove transactionPropTypes and MoneyRequestParticipantsSelector.

Deleted both files

**Comment by mountiny** [comment]
Ready for a review @hungvu193

**Comment by hungvu193** [comment]
Sure thing

File: contributingGuides/STYLE.md (hunk 4 above)
Comment (hungvu193) [reply contributingGuides/STYLE.md:117]: hmm ok, let be it.

File: src/components/OptionsSelector/BaseOptionsSelector.js [hunk 6]
```diff
@@ -496,8 +501,8 @@ class BaseOptionsSelector extends Component {
     render() {
         const shouldShowShowMoreButton = this.state.allOptions.length > CONST.MAX_OPTIONS_SELECTOR_PAGE_LENGTH * this.state.paginationPage;
         const shouldShowFooter =
-            !this.props.isReadOnly && (this.props.shouldShowConfirmButton || this.props.footerContent) && !(this.props.canSelectMultipleOptions && _.isEmpty(this.props.selectedOptions));
-        const defaultConfirmButtonText = _.isUndefined(this.props.confirmButtonText) ? this.props.translate('common.confirm') : this.props.confirmButtonText;
+            !this.props.isReadOnly && (this.props.shouldShowConfirmButton || this.props.footerContent) && !(this.props.canSelectMultipleOptions && this.props.selectedOptions.length === 0);
+        const defaultConfirmButtonText = typeof this.props.confirmButtonText === 'undefined' ? this.props.translate('common.confirm') : this.props.confirmButtonText;
```
Comment (hungvu193) [inline src/components/OptionsSelector/BaseOptionsSelector.js:505]: NAB, I think `this.props.confirmButtonText === 'undefined'` is enough.

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js [hunk 7]
```diff
@@ -1,8 +1,14 @@
+import lodashEvery from 'lodash/every';
 import lodashGet from 'lodash/get';
+import lodashIsEqual from 'lodash/isEqual';
+import lodashMap from 'lodash/map';
+import lodashPick from 'lodash/pick';
+import lodashReject from 'lodash/reject';
+import lodashSome from 'lodash/some';
+import lodashValues from 'lodash/values';
```
Comment (hungvu193) [inline src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js:8]: NAB, can we just use `import _ from 'lodash'` instead of importing every function here?

**Review by hungvu193 — COMMENTED** [review]
Looks good. I left few NAB comments/

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by hungvu193** [comment]
We had some conflicts again @bernhardoj

File: src/components/OptionsSelector/BaseOptionsSelector.js (hunk 6 above)
Comment (bernhardoj) [reply src/components/OptionsSelector/BaseOptionsSelector.js]: You're right, updated!

**Comment by bernhardoj** [comment]
@hungvu193 conflicts solved

File: tests/unit/CIGitLogicTest.sh [hunk 8]
```diff
@@ -77,7 +76,7 @@ function bump_version {
   setup_git_as_osbotify
   git switch main
   npm --no-git-tag-version version "$(ts-node "$bumpVersion" "$(print_version)" "$1")"
-  git add package.json package-lock.json
```
[the remark this answers is not in the document]
Comment (blazejkustra) [reply tests/unit/CIGitLogicTest.sh:80]: I think we shouldn't remove these two lines, maybe a better idea would be to install a different package?

```sh
  npm install lodash
```

File: contributingGuides/STYLE.md (hunk 4 above)
Comment (blazejkustra) [reply contributingGuides/STYLE.md:117]: We are going to adjust this markdown in a separate PR, so we can leave it as it is now 👍

File: contributingGuides/STYLE.md [hunk 9]
```diff
@@ -448,7 +419,7 @@ const propTypes = {
 
 ### Important Note:
 
-In React Native, one **must not** attempt to falsey-check a string for an inline ternary. Even if it's in curly braces, React Native will try to render it as a `<Text>` node and most likely throw an error about trying to render text outside of a `<Text>` component. Use `_.isEmpty()` instead.
+In React Native, one **must not** attempt to falsey-check a string for an inline ternary. Even if it's in curly braces, React Native will try to render it as a `<Text>` node and most likely throw an error about trying to render text outside of a `<Text>` component. Use `.length > 0` instead.
```
Comment (blazejkustra) [inline contributingGuides/STYLE.md:422]: Why not double negation?

```suggestion
In React Native, one **must not** attempt to falsey-check a string for an inline ternary. Even if it's in curly braces, React Native will try to render it as a `<Text>` node and most likely throw an error about trying to render text outside of a `<Text>` component. Use `!!` instead.
```

File: src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js (hunk 7 above)
Comment (blazejkustra) [reply src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js:8]: Yes we prefer importing like @bernhardoj did 👍

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: tests/unit/CIGitLogicTest.sh (hunk 8 above)
[the remark this answers is not in the document]
Comment (bernhardoj) [reply tests/unit/CIGitLogicTest.sh]: Ah, totally agree with that. Updated!

File: contributingGuides/STYLE.md (hunk 9 above)
Comment (bernhardoj) [reply contributingGuides/STYLE.md]: Oh, didn't think of that. I was just converting isEmpty to .length > 0.

**Review by blazejkustra — APPROVED** [review]

**Review by hungvu193 — APPROVED** [review]
All yours @mountiny

File: contributingGuides/STYLE.md (hunk 5 above)
Comment (mountiny) [reply contributingGuides/STYLE.md:117]: Should we still add this information in here? Seems like there is a preference and posibility of using lodash so it makes sense to leave a guide here in terms of what is preferred

File: contributingGuides/STYLE.md (hunk 4 above)
Comment (mountiny) [reply contributingGuides/STYLE.md:117]: ok

**Review by mountiny — APPROVED** [review]
thanks, great job on this one
